packages/aragon-cli/src/commands/dao_cmds/assign-id.js

```diff
@@ -0,0 +1,127 @@
+const TaskList = require('listr')
+const ENS = require('ethereum-ens')
+const { ensureWeb3 } = require('../../helpers/web3-fallback')
+const { green } = require('chalk')
+const listrOpts = require('@aragon/cli-utils/src/helpers/listr-options')
+const { isAddress } = require('web3-utils')
+const registrarAbi = require('@aragon/id/abi/IFIFSResolvingRegistrar').abi
+const { sha3 } = require('web3-utils')
+
+const ARAGON_DOMAIN = 'aragonid.eth'
+
+exports.command = 'assign-id <dao> <aragon-id>'
+exports.describe = 'Assign an Aragon Id to a DAO address'
+
+exports.builder = yargs => {
+  return yargs
+    .positional('dao', {
+      description: 'DAO address',
+      type: 'string',
+    })
+    .positional('aragon-id', {
+      description: 'Aragon Id',
+      type: 'string',
+    })
+}
+
+exports.task = async ({
+  dao,
+  aragonId,
+  web3,
+  gasPrice,
+  apmOptions,
+  silent,
+  debug,
+  reporter,
+}) => {
+  if (!isAddress(dao)) {
+    reporter.error('Invalid DAO address')
+    process.exit(1)
+  }
+
+  const tasks = new TaskList(
+    [
+      {
+        title: 'Validating Id',
+        task: async ctx => {
+          if (
+            !/^([\w-]+)$/.test(aragonId) &&
+            !new RegExp(`^([\\w-]+).${ARAGON_DOMAIN}$`).test(aragonId)
+          ) {
+            reporter.error('Invalid Aragon Id')
+            process.exit(1)
+          }
+
+          const ens = (ctx.ens = new ENS(
+            web3.currentProvider,
+            apmOptions['ens-registry']
+          ))
+
+          ctx.domain = aragonId.includes(ARAGON_DOMAIN)
+            ? aragonId
+            : `${aragonId}.${ARAGON_DOMAIN}`
+
+          // Check name doesn't already exist
+          try {
+            const exists = await ens.resolver(ctx.domain).addr()
+
+            if (exists) {
+              reporter.error(
+                `Cannot assign: ${ctx.domain} is already assigned to ${exists}.`
+              )
+              process.emit(1)
+            }
+          } catch (err) {
+            // ens.resolver() throws an ENS.NameNotFound error if name doesn't exist
+            if (err !== ENS.NameNotFound) throw err
+          }
+        },
+      },
+      {
+        title: 'Assigning Id',
```

Comment:
I notice this task is executed even when id can't be assigned. Maybe we can pass `exists` throught `ctx` and only enable the task if `!ctx.exists`?

Reply:
Nice catch! The problem is caused by a silly typo. It should be `process.exit(1)` instead of `process.emit(1)` on line `72`.